src/main/java/juja/microservices/keepers/service/KeepersService.java

```diff
@@ -64,6 +64,22 @@ public String addKeeper(KeeperRequest keeperRequest){
     }
 
     public Map<String, List<String>> getActiveKeepers() {
-        return keepersRepository.getActiveKeepers();
+        logger.debug("Service.getActiveKeepers after in, without any parameters.");
+        Map<String, List<String>> outMap = new HashMap<>();
+        List<Keeper> keepers = keepersRepository.getActiveKeepers();
+        for (Keeper keeper : keepers) {
+            List<String> directions = new ArrayList<>();
+            if(outMap.containsKey(keeper.getUuid())) {
+                directions = outMap.get(keeper.getUuid());
```

Comment:
You invoke several times methods keeper.getUuid() and keeper.getDirection() in this loop. You can in the begining of this loop get these methods result and asign to the variables. Then use these variables.